src/asset-management/AaveManager.sol

```diff
@@ -96,39 +98,38 @@ contract AaveManager is IAssetManager, Ownable, ReentrancyGuard
 
         // withdraw from the market
         if (aAmount0Change < 0) {
-            _doDivest(aPair, lToken0, uint256(-aAmount0Change));
+            _doDivest(aPair, lToken0, lToken0AToken, uint256(-aAmount0Change));
         }
         if (aAmount1Change < 0) {
-            _doDivest(aPair, lToken1, uint256(-aAmount1Change));
+            _doDivest(aPair, lToken1, lToken1AToken, uint256(-aAmount1Change));
         }
 
         // transfer tokens to/from the pair
         aPair.adjustManagement(aAmount0Change, aAmount1Change);
 
         // transfer the managed tokens to the destination
         if (aAmount0Change > 0) {
-            _doInvest(aPair, lToken0, uint256(aAmount0Change));
+            _doInvest(aPair, lToken0, lToken0AToken, uint256(aAmount0Change));
         }
         if (aAmount1Change > 0) {
-            _doInvest(aPair, lToken1, uint256(aAmount1Change));
+            _doInvest(aPair, lToken1, lToken1AToken, uint256(aAmount1Change));
         }
     }
 
-    function _doDivest(IAssetManagedPair aPair, IERC20 aToken, uint256 aAmount) private {
-        uint256 lShares = _updateShares(aPair, address(aToken), aAmount, false);
+    function _doDivest(IAssetManagedPair aPair, IERC20 aToken, address aAaveToken, uint256 aAmount) private {
```

Comment:
this is for perf purposes. To save another external call when we have already queried the address of the AaveToken